src/XrmMockupWorkflow/WorkflowTree.cs

```diff
@@ -1301,9 +1301,30 @@ public Assign(string To, string Value)
             if (Value.Contains(".Id"))
             {
                 var toEntity = variables[To.Replace(".Id", "")] as Entity;
-                var valueEntity = variables[Value.Replace(".Id", "")] as Entity;
-                toEntity.Id = valueEntity.Id;
-                return;
+                if (Value.Contains("related_"))
```

Comment:
There is no test case for this change. Do you have a workflow that reaches this code?

Reply:
I will do somewhere, but I can't remember which one it is, and as it'll be in my live project it will be under an NDA anyway. TBH this change seems unrelated to the rest of the PR so perhaps exclude this bit and I'll come back to it ?

Reply:
Hi @magesoe 
PR #173 has the proper fix for the workflow issue.